app/com/arpnetworking/metrics/portal/scheduling/JobRepository.java

```diff
@@ -53,14 +53,16 @@
     Optional<Job<T>> getJob(UUID id, Organization organization);
 
     /**
-     * Get the last time that a job with a given UUID was run.
+     * Get the last time that any completed execution for a given job was scheduled for.
+     *
+     * More precisely, in pseudocode, {@code job.executions().filter(hasCompleted).map(scheduledFor).max()}.
      *
      * @param id The id assigned to the Job by a previous call to {@code add}.
      * @param organization The organization owning the job.
-     * @return The last time that that job was executed.
+     * @return The last time that any completed execution for the job was scheduled for.
      * @throws NoSuchElementException if no job has the given UUID.
      */
-    Optional<Instant> getJobLastRun(UUID id, Organization organization) throws NoSuchElementException;
+    Optional<Instant> getJobLastExecutionScheduled(UUID id, Organization organization) throws NoSuchElementException;
```

Comment:
Naming: Seems somewhat ambiguous in the sense that it is for the last completed job; although I get how the other was misleading as well, that it returns the scheduled date and not the run (start/end) date/time.

Perhaps: `getJobLastRunScheduledAt` -- not a fan of "run", but I want the name to convey that its the scheduled date time for a job that has completed (success or failure); `getMostRecentJobCompletedScheduledDate` (wow that's a mouthful) ... `getLastJobCompletedScheduledDate` ... sigh naming sucks.

Reply:
Yeah, I think that this concept is inherently high-Kolmogorov-complexity, so any name must be either verbose or ambiguous. I tend to prefer "verbose" to "ambiguous," and would lean towards `getLastScheduledTimeWhereExecutionCompleted`, but I could also do `getMostRecentJobCompletedScheduledDate` as you suggest.